### DISCUSSION TITLE
Request Response

### DISCUSSION THREAD

**Discussion by ChipHosting** [AshleyRedman/WPClothes2Order#95] [Q&A] [answered]
Hi,

Is there somewhere to see the request response sent to Clothes 2 order? I set a product embroidery to 10cm but they have said it has been sent as 1cm. It would be good to be able to check the request sent to debug. 

Also my image optimiser (shrinking) plugin may be impacting the upload so would be good to see what image was sent so I can somehow make sure its the full sized image sent.

Thanks in advance :)

**Comment by AshleyRedman** [comment, marked as answer]
Hey @adamcook102, I put something in the plugin options to see what an example request might look like ([API tab here](https://www.wpclothes2order.com/_next/image?url=%2Fscreenshots%2Fscreenshot-3.png&w=828&q=75)), however looking at your situation, i think it may be worth allowing users to see the actual full request in their order list/history ([Order history tab](https://www.wpclothes2order.com/_next/image?url=%2Fscreenshots%2Fscreenshot-3.png&w=828&q=75)).

Would that help with your issue? The full JSON request that is sent etc

**Reply by ChipHosting** [reply]
That would be perfect, even if its a toggle to enable when debugging. Seeing the request/ response when setting it up would be perfect to see the exact details sent.

Thanks

**Reply by AshleyRedman** [reply]
@adamcook102 No problem, I will open an issue & look to get this deployed as a new minor patch this week. Thanks for the suggestion

**Reply by AshleyRedman** [reply]
Please keep an eye on issue: https://github.com/AshleyRedman/WPClothes2Order/issues/96

**Comment by AshleyRedman** [comment]
@adamcook102 Couldn't help myself, release version `1.0.5` which now includes this. From within your WordPress Admin, navigate to `Order History` & you will see a new button per each order which will now show you the payload sent to C2O. This request is now stored in the database, so only new orders will show their requests. Hope this helps!

![Screenshot 2022-12-05 at 22 42 30](https://user-images.githubusercontent.com/7602057/205763355-60560cac-98f5-4f46-8c87-38ebdc0e9008.png)

**Reply by ChipHosting** [reply]
Amazing thank you will play around tomorrow 🥇

**Reply by ChipHosting** [reply]
I also couldn't help myself, updated but it broke my site so I've had to downgrade back to an older version. Was just getting errors around request and response files.

**Reply by AshleyRedman** [reply]
Interesting, would you mind sharing some info on the errors you are getting?

**Reply by AshleyRedman** [reply]
Hi @ChipHosting I have recently release `1.0.7` which resolved a bug with stock syncing, as well as updating this feature mentioned to also include the raw response from C2O. I have tested on a live site & seems to be working as expected. So closing off this now, however feel free to open a new if you are still having issues.
